HomeAssistant.py

```diff
@@ -678,28 +687,25 @@ def addSynonyms(self) -> bool:
 	def processHADataRetrieval(self):
 		# extra method to reduce complexity value of addHomeAssistantDevices()
 		# clean up any duplicates in the list
-		duplicateList = set(tuple(x) for x in self._switchAndGroupList)
-		finalList = [list(x) for x in duplicateList]
+		# duplicateList = set(tuple(x) for x in self._switchAndGroupList)
+		duplicateList = dict((x[0], x) for x in self._switchAndGroupList).values()
 
```

Comment:
You're putting tuples in a dictionary without keys, right? Can't test right now, just looks odd, might be correct though

Reply:
yes, it seems to be a effective way to check the first item in the list is not a duplicate in the remaining nested lists

Reply:
An example  payload for this is.... [deviceId, State, Friendly name]..... EG: [['switch.desk_lamp', 'ON', 'desk lamp'],[ ['switch.garage_light', 'OFF', 'garage light']] .